subsys/mgmt/mcumgr/grp/os_mgmt/src/os_mgmt_b0_active_slot.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zcbor_common.h>
+#include <zcbor_encode.h>
+#include <pm_config.h>
+#include <fw_info.h>
+#include <zephyr/mgmt/mcumgr/mgmt/callbacks.h>
+#include <zephyr/mgmt/mcumgr/grp/os_mgmt/os_mgmt.h>
+
+static enum mgmt_cb_return bootloader_info_hook(uint32_t event, enum mgmt_cb_return prev_status,
+						int32_t *rc, uint16_t *group, bool *abort_more,
+						void *data, size_t data_size)
+{
+	struct os_mgmt_bootloader_info_data *bootloader_info_data =
+						(struct os_mgmt_bootloader_info_data *)data;
+
+	if (event != MGMT_EVT_OP_OS_MGMT_BOOTLOADER_INFO || data_size !=
+	    sizeof(*bootloader_info_data)) {
+		*rc = MGMT_ERR_EUNKNOWN;
+		return MGMT_CB_ERROR_RC;
+	}
+
+	if (*(bootloader_info_data->decoded) >= 1 && (sizeof("active_b0_slot") - 1) ==
+	    bootloader_info_data->query->len && memcmp("active_b0_slot",
+						       bootloader_info_data->query->value,
+						       bootloader_info_data->query->len) == 0) {
+		const struct fw_info *s0_info = fw_info_find(PM_S0_ADDRESS);
+		const struct fw_info *s1_info = fw_info_find(PM_S1_ADDRESS);
+
+		if (s0_info || s1_info) {
+			uint32_t active_slot;
+			bool ok;
+
+			if (!s1_info || (s0_info && s0_info->version >= s1_info->version)) {
+				active_slot = 0;
+			} else if (!s0_info || (s1_info && s1_info->version > s0_info->version)) {
+				active_slot = 1;
+			}
```

Comment:
Assuming that at this point we can have both `s0_info` and `s1_info` and the above `else` and `else if` have not been triggered, with the `!s0_info` and `!s1_info,` we have no processing of situation where both `s0_info && s1_info`, in which case `active_slot` is left uninitialized.

Reply:
there must be an s0_info or s1_info for b0 to be able to boot the image so one of the conditions must run, if there is no s0_info and no s1_info and b0 has booted the image then something is very wrong

Reply:
Right, my bad.